src/dotless.Core/Parser/Tree/Ruleset.cs

```diff
@@ -279,6 +281,50 @@ protected void EvaluateRules(Env env)
             env.Frames.Pop();
         }
 
+        /// <summary>
+        /// Merges properties with merge operators '+' and '+_'
+        /// </summary>
+        /// <returns>New merged ruleset</returns>
+        private NodeList MergeProperties()
+        {
+            var groups = new Dictionary<string, List<Rule>>();
+            var sorted = new List<Node>();
+            foreach (var node in Rules)
+            {
+                var rule = node as Rule;
+                if (rule == null || rule.Merge == "")
+                {
+                    sorted.Add(node);
+                    continue;
+                }
+                var key = $"{rule.Name},{rule.Important},{rule.Merge}";
+                if (!groups.ContainsKey(key))
+                {
+                    groups[key] = new List<Rule> {rule};
+                    sorted.Add(node);
+                }
+                else
+                {
+                    if (rule.Merge == "")
+                        sorted.Add(rule);
+                    else
+                        groups[key].Add(rule);
+                }
+            }
+            foreach (var group in groups.Values)
+            {
+                var rule = @group[0];
+                var value = rule.Value as Value;
+                if (value == null)
+                    value = (Value) (rule.Value = new Value(new[] {rule.Value}, "", rule.Merge));
```

Comment:
These couple of lines here really need some clarification while you still remember what's going on. It's not immediately obvious that this seems to cover a case where Rule.Value is not a Value. Also, while I love terse code, this line in particular has way too many meanings for the word Value to be readable, so maybe expand it a little bit? :)